src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1401 @@ const ActorInit En_Nb_InitVars = {
+            this->actor.flags |= 9;
+            this->actor.posRot.pos = this->finalPos;
+            this->action = NB_IDLE_AFTER_TALK;
+            this->drawMode = NB_DRAW_DEFAULT;
+        }
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void func_80AB359C(EnNb* this) {
+    PosRot* posRot = &this->actor.posRot;
+    Vec3f* initialPos = &this->initialPos;
+    Vec3f* finalPos = &this->finalPos;
+    f32 f0;
+    u16 temp_t1;
+    s16 temp_2;
+
+    this->movementTimer++;
+    temp_2 = kREG(17);
+    temp_t1 = temp_2;
+    temp_t1 += 0x19; // Oddly needs all these temps to match
+    if (temp_t1 >= this->movementTimer) {
+        f0 = func_8006F9BC(temp_t1, 0, this->movementTimer, 3, 3);
+        posRot->pos.x = initialPos->x + (f0 * (finalPos->x - initialPos->x));
+        posRot->pos.y = initialPos->y + (f0 * (finalPos->y - initialPos->y));
+        posRot->pos.z = initialPos->z + (f0 * (finalPos->z - initialPos->z));
+    }
+}
+
+void EnNb_SetNoticeSFX(EnNb* this) {
+    func_80078914(&this->actor.projectedPos, NA_SE_VO_NB_NOTICE);
+}
+
+s32 EnNb_GetNoticedStatus(EnNb* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 playerX = player->actor.posRot.pos.x;
+    f32 playerZ = player->actor.posRot.pos.z;
+    f32 thisX = this->actor.posRot.pos.x;
+    f32 thisZ = this->actor.posRot.pos.z;
+
+    if (SQ(playerX - thisX) + SQ(playerZ - thisZ) < 6400.0f) {
+        return true;
+    }
+    return false;
+}
+
+void func_80AB36DC(Actor* thisx, GlobalContext* globalCtx) {
+    EnNb* this = THIS;
+    u16 moveTime = this->movementTimer;
+    s16* shapeRot = &this->actor.shape.rot.y;
+
+    if ((((u16)((u16)(kREG(17) + 0x19) - 4))) > moveTime) {
+        s16 invScale = 4 - moveTime;
+        if (invScale > 0) {
+            Math_SmoothScaleMaxMinS(shapeRot, this->pathYaw, invScale, 6200, 100);
+        }
+    } else {
+        s16 invScale = (u16)(kREG(17) + 0x19) - moveTime;
+        if (invScale > 0) {
+            Math_SmoothScaleMaxMinS(shapeRot, this->actor.initPosRot.rot.y, invScale, 6200, 100);
+        }
+    }
+}
+
+void EnNb_CheckNoticed(EnNb* this, GlobalContext* globalCtx) {
+    if (EnNb_GetNoticedStatus(this, globalCtx)) {
+        EnNb_SetCurrentAnim(this, &D_06004E60, 2, -8.0f, 0);
+        this->action = NB_NOTICE_PLAYER;
+        EnNb_SetNoticeSFX(this);
+    }
+}
+
+void EnNb_SetupIdleCrawlspace(EnNb* this, s32 animFinished) {
+    if (animFinished) {
+        EnNb_SetCurrentAnim(this, &D_06004BB4, 0, -8.0f, 0);
+        this->headTurnFlag = 1;
+        this->actor.flags |= 9;
+        this->action = NB_IDLE_CRAWLSPACE;
+    }
+}
+
+void func_80AB3838(EnNb* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->action = NB_IN_DIALOG;
+    } else {
+        this->actor.flags |= 9;
+        if (!(gSaveContext.infTable[22] & 0x1000)) {
+            this->actor.textId = 0x601D; // "I haven't seen you around, kid...What do you want?"
```

Comment:
we don't put text excerpts anymore [apparently](https://github.com/zeldaret/oot/pull/380#discussion_r491707947)